harness/determined/pytorch/_lightning/adapter.py

```diff
@@ -0,0 +1,209 @@
+        super().__init__(context)
+        check_compat(lightning_module)
+        self.lm = lightning_module
+        self.context = context
+
+        optimizers, lr_schedulers = self.setup_optimizers_schedulers(context)
+        self.optimizers = optimizers
+
+        # user should merge these with their own callbacks
+        self.build_callbacks = lambda: build_callbacks(lightning_module, self.context)
+
+        # set lightning_module properties
+        self.lm.use_ddp = False
+        self.lm.use_ddp2 = False
+        self.lm.use_dp = False
+        self.lm.use_tpu = False
+        self.lm.local_rank = self.context.distributed.get_local_rank()
+        self.lm.global_rank = self.context.distributed.get_rank()
+        self.lm.use_amp = self.context._use_amp
+        self.lm.to(self.context.device)
+
+    def setup_optimizers_schedulers(
+        self, context: PyTorchTrialContext
+    ) -> Tuple[List[Optimizer], List[_LRScheduler]]:
+        """
+        Wrap optimizers and lr_schedulers returned by `configure_optimizers` to
+        work with Determined.
+        Return: Wrapped `optimizers`, and `lr_schedulers` in a tuple
+        """
+        optimizers, lr_scheduler_dicts, opt_frequencies = ptl_utils.init_optimizers(self.lm)
+        for freq in opt_frequencies:
+            check.eq(freq, 1, "custom optimizer frequencies are not supported")
+        optimizers = cast(List[Optimizer], optimizers)
+        lr_scheduler_dicts = cast(List[dict], lr_scheduler_dicts)
+
+        def lightning_scheduler_dict_to_det(lrs: dict) -> _LRScheduler:
+            """
+            input_dict = {
+                'scheduler': None,
+                'name': None,  # no custom name
+                'interval': 'epoch',  # after epoch is over
+                'frequency': 1,  # every epoch/batch
+                'reduce_on_plateau': False,  # most often not ReduceLROnPlateau scheduler
+                'monitor': monitor,  # value to monitor for ReduceLROnPlateau
+                'strict': True,  # enforce that the monitor exists for ReduceLROnPlateau
+            }
+            """
+            # TODO support custom frequencies with the manual step.
+            step_mode = (
+                LRScheduler.StepMode.STEP_EVERY_EPOCH
+                if lrs["interval"] == "epoch"
+                else LRScheduler.StepMode.STEP_EVERY_BATCH
+            )
+            return context.wrap_lr_scheduler(lrs["scheduler"], step_mode)
+
+        optimizers = [self.context.wrap_optimizer(opt) for opt in optimizers]
+        lr_schedulers = [lightning_scheduler_dict_to_det(lrs) for lrs in lr_scheduler_dicts]
+        return optimizers, lr_schedulers
+
+    def train_batch(
+        self, batch: TorchData, epoch_idx: int, batch_idx: int
+    ) -> Union[torch.Tensor, Dict[str, Any]]:
+        self.lm.on_train_batch_start(batch, batch_idx, dataloader_idx=0)
+        type(self.lm).global_step = mock.PropertyMock(return_value=batch_idx)
```

Comment:
I tried `type(self.lm).global_step=batch_idx`. It worked.
 maybe we don't need this mock package?